modules/transactionpool/update.go

```diff
@@ -33,17 +36,87 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 				tp.log.Println("ERROR: could not delete a transaction:", err)
 			}
 		}
+
+		// Pull the transactions out of the fee summary. For estimating only
+		// over 10 blocks, it is extermely likely that there will be more
+		// applied blocks than reverted blocks, and if there aren't (a height
+		// decreasing reorg), there will be more than 10 applied blocks.
+		if len(tp.txnsPerBlock) > 0 {
+			tailOffset := uint64(0)
+			for i := 0; i < len(tp.txnsPerBlock)-1; i++ {
+				tailOffset += tp.txnsPerBlock[i]
+			}
+
+			// Strip out all of the transactions in this block.
+			tp.recentConfirmedFees = tp.recentConfirmedFees[:tailOffset]
+			// Strip off the tail offset.
+			tp.txnsPerBlock = tp.txnsPerBlock[:len(tp.txnsPerBlock)-1]
+		}
```

Comment:
This feels like a lot of code for it's effective function - pop the most recent block off of the list of blocks.